src/librustc_infer/infer/mod.rs

```diff
@@ -892,13 +892,18 @@ impl<'tcx> InferOk<'tcx, ()> {
 }
 
 #[must_use = "once you start a snapshot, you should always consume it"]
-pub struct CombinedSnapshot<'a, 'tcx> {
-    undo_snapshot: Snapshot<'tcx>,
+pub struct FullSnapshot<'a, 'tcx> {
```

Comment:
Comment: What is the difference between a "full snapshot" and a "combined snapshot"? When would I use one vs the other?